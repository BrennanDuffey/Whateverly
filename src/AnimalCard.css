.cards {
 border: dotted 2px grey;
 justify-content: center;
 margin: auto;
 width: 50%;
}

img {
  width: 100%;
}

p {
  text-align: left;
  font-weight: bold;
  margin: 10px;
}

