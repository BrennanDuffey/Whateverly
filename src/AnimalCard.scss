.cards {
 object-fit: contain;
 background-color: #102731F2; 
 box-shadow: 5px 15px 7px rgba(0, 0, 0, .5);
 justify-content: center;
 margin: 30px auto;
 width: 50%;
}

article {
  padding: 5px;
}

img {
  height: 172.8px;
  object-fit: cover;
  width: 100%;
}

button {
  font-family: Apple Chancery;
  font-size: 1rem;
  height: 28px;
  width: 35%;
}

.favorited {
  background:rgb(204, 190, 68);
  font-weight: bolder;
}

h2 {
  margin: 2px;
  color: #FCFDFF;
  font-size: 3.5em;
}

p {
  text-align: left;
  margin: 10px;
}

.animal-info-red {
  color: #FF6232;
  font-weight: 400;
  font-size: 1.2rem;
}
.animal-info {
  color: #1A657C;
  font-weight: 400;
  font-size: 1.2rem;

}
.animal-info-titles {
  color: #8FA7AD;
  font-weight: bold;
  font-size: 1.5rem;

}

