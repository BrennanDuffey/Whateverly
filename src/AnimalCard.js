import React, { Component } from 'react';
import './AnimalCard.scss';

class AnimalCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: props.name,
      isFav: false
    }
  }
  render () {
    let favbtn;
    let favClass = "";
    if(this.props.favAnimals.includes(this.props.name)) {
      favbtn = 'Favorited!'
      favClass = ' favorited'
      // console.log('name: ', props.favAnimals.includes(props.name))
    } else{
      favbtn = 'Add to Favorites'
    }
    return (
      <section className="cards">
        <img src={this.props.image} />
        <article>
          <button onClick={(e) => {
          this.props.addFav(this.props.name)
          
        }
          } id='favbtn' className={favClass}>{favbtn}</button>

        <h2>{this.props.name}</h2>
        <p className="animal-info-titles">Extinction Level: <span className="animal-info-red">{this.props.endangeredStatus}</span></p>
        <p className="animal-info-titles">Population: <span className="animal-info-red">{this.props.population}</span></p>
        <p className="animal-info-titles">Genus: <span className="animal-info">{this.props.genus}</span></p>
        <p className="animal-info-titles">Locations: <span className="animal-info">{this.props.locations.join(', ')}</span></p>
        <p className="animal-info-titles">Threats: <span className="animal-info">{this.props.threats.join(', ')}</span></p>
        </article>
      </section>
    )
  }
}

export default AnimalCard; 