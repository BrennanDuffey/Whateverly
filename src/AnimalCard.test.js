import React from 'react';
import AnimalCard from './AnimalCard.js';
import { shallow } from 'enzyme';


const mockEndangeredSpecies = 
    {
    "name": "Snow Leopard",
    "genus": "Panthera",
    "endangeredStatus": "Vulnerable",
    "locations": ["China", "Mongolia", "India"],
    "population": 6000,
    "threats": ["Poaching","Climate Change"],
    "image": "https://upload.wikimedia.org/wikipedia/commons/d/d7/Snow_Leopard_in_Ladakh%28_Photo_by_Tashi_Lonchay%29.jpg"
    };
const mockAddFav = jest.fn();
const mockSearchInput = 'Jaguar';
const mockFavAnimals = ['Jaguar', 'Narwhal'];
const mockDefaultProps = props;

describe('AnimalCard', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <AnimalCard {...mockEndangeredSpecies}
                        addFav={mockAddFav}
                        searchInput={mockSearchInput}
                        favAnimals={mockFavAnimals}
            />
        )
    });
    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    }); 
    it('should have a proper default state', () => {
        expect(wrapper.state()).toEqual( {isFav: false} )
    });
});