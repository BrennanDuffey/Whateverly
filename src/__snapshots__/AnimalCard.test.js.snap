// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimalCard should match the snapshot 1`] = `
<section
  className="cards"
>
  <img
    src="https://upload.wikimedia.org/wikipedia/commons/d/d7/Snow_Leopard_in_Ladakh%28_Photo_by_Tashi_Lonchay%29.jpg"
  />
  <article>
    <button
      className=""
      id="favbtn"
      onClick={[Function]}
    >
      Add to Favorites
    </button>
    <h2>
      Snow Leopard
    </h2>
    <p
      className="animal-info-titles"
    >
      Extinction Level: 
      <span
        className="animal-info-red"
      >
        Vulnerable
      </span>
    </p>
    <p
      className="animal-info-titles"
    >
      Population: 
      <span
        className="animal-info-red"
      >
        6000
      </span>
    </p>
    <p
      className="animal-info-titles"
    >
      Genus: 
      <span
        className="animal-info"
      >
        Panthera
      </span>
    </p>
    <p
      className="animal-info-titles"
    >
      Locations: 
      <span
        className="animal-info"
      >
        China, Mongolia, India
      </span>
    </p>
    <p
      className="animal-info-titles"
    >
      Threats: 
      <span
        className="animal-info"
      >
        Poaching, Climate Change
      </span>
    </p>
  </article>
</section>
`;
